kimchi_state/src/navigation_manager.cpp

```diff
@@ -68,8 +70,67 @@ void NavigationManager::startNavigation() {
 
 void NavigationManager::stopNavigation() {}
 
+void NavigationManager::localizeGoalResponseCallback(
+  GoalHandleGlobalLocalization::SharedPtr goal_handle) {
+    if (!goal_handle) {
+      RCLCPP_ERROR(node_->get_logger(), "[NavigationManager] Goal was rejected by server");
+    } else {
+      RCLCPP_INFO(node_->get_logger(), "[NavigationManager] Goal accepted by server, waiting for "
+                                      "feedback and result...");
+    }
+}
+
+void NavigationManager::localizeFeedbackCallback(
+  GoalHandleGlobalLocalization::SharedPtr,
+  const std::shared_ptr<const GlobalLocalization::Feedback> feedback) {
+    RCLCPP_DEBUG(node_->get_logger(),
+                    "[NavigationManager] Received feedback: Current pose estimate - x: %.2f, y: %.2f",
+                    feedback->pose_feedback.pose.pose.position.x,
+                    feedback->pose_feedback.pose.pose.position.y);
+}
+
+void NavigationManager::localizeResultCallback(const GoalHandleGlobalLocalization::WrappedResult &result) {
```

Comment:
Is there a way of letting us know why it failed? Again, having the variance here might be helpful